templates/etc/td-agent/td-agent.conf

```diff
@@ -28,10 +28,17 @@
   buffer_chunk_limit 1M
   # Flush logs every 5 seconds, even if the buffer is not full.
   flush_interval 5s
-  # Never wait longer than 5 minutes between retries.
+  # Enforce some limit on the number of retries.
+  disable_retry_limit false
+  # The maximum number of retries for sending a chunk. After this many of
+  # retries, the given chunk will be discarded.
+  retry_limit 8
+  # Wait 1 second before the first retry. The wait interval will be doubled on
+  # each following retry (2s, 4s, 8s...) until it reaches the max_retry_wait.
+  retry_wait 1
+  # Never wait longer than 5 minutes between retries. If the wait interval
+  # reaches this limit, the exponentiation stops.
   max_retry_wait 300
```

Comment:
What's the relationship between retry_limit and max_retry_wait?

If retry_limit is 8 and all 8 retries are needed, the time between the 7th and 8th retry would be 64s, right? Given that this is less than max_retry_wait of 300, do we still need/want to set the max_retry_wait parameter?

Reply:
Exactly! I've just removed `max_retry_wait` since we will never hit it.

Reply:
The default for `max_retry_wait` is unset, which means keep doubling forever. I'm worried that someone will adjust the `retry_limit` and `retry_wait` values in the default config and forget to set `max_retry_wait`, and then wonder why the agent isn't sending a log entry for 24 hours (true story). Would it make sense to keep it in the config, but comment that with the above settings we should never reach it?

Reply:
Yep I had this worry too.

Reply:
Added it back with a comment to clarify.